Toolkit.py

```
@@ -1,321 +1,322 @@
+		logger.debug(f"ChangeConsoleStyle() Run Time: {(time.perf_counter_ns() - startTime)/1000000} ms");
+
+	def GetHash(filePath:str=None)->str:
+		startTime:int = time.perf_counter_ns();
+		if (filePath is None): return "";
+		with open(filePath,'rb') as fileBytes:
+			hashResult:str = sha256(fileBytes.read()).hexdigest().upper();
+		logger.debug(f"GetHash() Run Time: {(time.perf_counter_ns() - startTime)/1000000} ms");
+		return hashResult;
+
+	@classmethod
+	def ResourceReleases(cls, offset:int, lenth:int, releasePath:str=None)->bytes:
```

Ruff (0.14.8)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)